slsim/Sources/source_variability/variability.py

```diff
@@ -42,3 +43,22 @@ def sinusoidal_variability(t, amp, freq):
     :return: variability for the given time
     """
     return amp * np.sin(2 * np.pi * freq * t)
+
+def interpolate_variability(Movie, Orig_timestamps, New_timestamps):
+    """Interpolates a variable source to any given time series
```

Comment:
Also, check why pre-commit is failing and fix them.

Reply:
It looks like pre-commit was failing because the lines were too long and I had superfluous spacings to align the comments. 

Now, when I run "pre-commit run --show-diff-on-failure --color=always --all-files" locally, I receive all 5 passes, but on Github I'm receiving an error for the hook "black" failing: 
ruff.....................................................................Passed
nbqa-ruff................................................................Passed
black....................................................................Failed
- hook id: black
- files were modified by this hook

reformatted slsim/Sources/source_variability/variability.py
reformatted tests/test_Source/test_source_variability.py

All done! ✨ 🍰 ✨
2 files reformatted, 54 files left unchanged.

black-jupyter............................................................Passed
docformatter.............................................................Passed

 As for why the function needs to be here, I guess it's because it is applicable to any series of screen shots, whether it be AGN or supernovae. Is there a variable class I should put it so it may be accessed cleaner?

Reply:
Hi @Henry-Best-01 , I am hoping that you have a plan to implement your quasar variability model.

Reply:
Yes I do! It would be silly for me to only contribute only a few lines of code.

I will open a new pull request when I have all the pieces of my black box in place.

Reply:
Added a sentence that this will be important for future microlensing implimentation